### ISSUE TITLE
Misc bugs and feedback with the latest version

### ISSUE THREAD

**Issue by Teacyn** [EtherealEmil/UsualScrap#6]
I've tested the mod for about an hour off and on checking everything I could immediately think of, though don't consider this to be a fully extensive test.

# Bugs
- Clicking with the toolbox while not looking at a turret/landmine causes the following error. This does not seem to actually do anything, but.

```
[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
UsualScrap.Behaviors.ToolboxFunction.ItemActivate (System.Boolean used, System.Boolean buttonDown) (at <f06a5dc17ec84cbea5e6f91f3628668a>:IL_0069)
GrabbableObject.UseItemOnClient (System.Boolean buttonDown) (at <255c0877f9414c3aa6f1fe5079ac13f3>:IL_005C)
(wrapper dynamic-method) GameNetcodeStuff.PlayerControllerB.DMD<GameNetcodeStuff.PlayerControllerB::ActivateItem_performed>(GameNetcodeStuff.PlayerControllerB,UnityEngine.InputSystem.InputAction/CallbackContext)
UnityEngine.InputSystem.Utilities.DelegateHelpers.InvokeCallbacksSafe[TValue] (UnityEngine.InputSystem.Utilities.CallbackArray`1[System.Action`1[TValue]]& callbacks, TValue argument, System.String callbackName, System.Object context) (at <dbb0ce7b96144bd8bcbcb8b0d765f970>:IL_001B)
UnityEngine.InputSystem.LowLevel.<>c__DisplayClass7_0:<set_onUpdate>b__0(NativeInputUpdateType, NativeInputEventBuffer*)
UnityEngineInternal.Input.NativeInputSystem:NotifyUpdate(NativeInputUpdateType, IntPtr)

[Error  : Unity Log] NullReferenceException while executing 'performed' callbacks of 'Movement/ActivateItem[/Mouse/leftButton]'
```

# Other issues/feedback

- Everything lacks a pick up and put down sound effect. I might suggest just reusing some of the vanilla ones if you don't have the ability to make your own, I.E you could use the V-type engine sound for the explosive tank, the big bolt for the Toolbox, plastic fish for medkit, etc. Might be stuff on freesound.org that fits too.
- Items still lack usage sounds. GeneralImprovements has a setting to add a medkit to the ship which has a sound that would make sense for the bandage, I'm unsure if that sound is custom or if it's a vanilla sound reused. Unsure what would fit the medkit though. Handlamp could probably just use the flashlight click sound, toolbox some kind of "mechanic work" noise that plays on a slow loop while using it? Unsure.
- Prior issue of the handlamp transforming into a flashlight if toggled while not using it persists. Would it be possible to simply disable it from working in the "off-hand" like flashlights can?
- While I'm aware the toolbox is intended to get sounds later to indicate it's working, I suggest reverting the mashing to press and hold for accessibility reasons. Clicking 30+ times as fast as I can is rough on the hands. If the issue is that press and hold makes the toolbox too consistent, could you possibly do a % chance for the interaction to fail on completion?
- Handlamp is in a good state now 👍 
- The Radioactive Cell is still called the mineral cell while the changelog says it was renamed. 
![image](https://github.com/EtherealEmil/UsualScrap/assets/12898157/1cdc538d-2556-4e2b-9d14-dbc66d59b3da)

- Great changes to the medkit model, looks way nicer. All of the new models are really nice,
- I've noticed in prior versions that the medkit had a bad tendency to cause the "throw items" glitch, where the game perpetually thinks you're in a state of throwing or using an item and you become unable to interact with anything. I have so far been unable to replicate this on this version, but I have not done extensive testing nor was I able to nail down a consistent cause on prior versions (I would have made a report if I did)

Overall a really nice update, glad to see ya back. Looking forward to the new stuff you're working on. I think the sound issues are about the only major thing I've still noticed that throw me off.

**Comment by EtherealEmil** [comment]
The error is harmless but I'll see if I can appease the code's demands anyway.
There are actually no sounds whatsoever because I had wanted to use the vanilla sounds but I also didn't know how to go about correctly accessing and using the vanilla sounds without it being considered stealing or whatever. I'll figure something out, I know having no sounds at all isn't ideal. 
The handlamp just uses the flashlight script with some tweaks so there are some things (The head light!) I cannot access or change unless I make a new script entirely, will do. 
The toolbox I'll consider reverting or maybe just lowering the amount of clicks to a smaller pool we'll see.
The cell name change when scanned just escaped making into my final submission like something always does :)
I experienced this throw glitch the medkit had with my buds too which is why I tried to fix it, I can't test it anytime soon but hopefully I got it.

Thanks for the help, I'll leave this open.

**Comment by Teacyn** [comment]
Found an additional bug. Explosive tanks seem to teleport slightly behind you when dropped instead of dropping like normal scrap do.

**Comment by EtherealEmil** [comment]
- After some testing I've noticed objects tend to drop in different spots based on your camera so if you look down objects tend to drop closer to your body. In the explosive tank's case the object is just long so if you look down the bottom of the tank will be through your body and then when dropped it falls back there. Not a lot I can do to fix that but I did shift the holding position a bit farther from your body to try and alleviate it.
- I think you are referring to the lightsout mod by mrov? The mod says it turns off lamps and the apparatus and added compatibility for another mod's lamps but I'd assume it targets those specific items and not everything with a light. I think my mod would need to be made compatible? Regardless, I took a quick glance at the mod and made some changes that may fix it they do work together, if there's another mod that does this I can check it out.
